src/components/cc-tile-scalability/cc-tile-scalability.js

```diff
@@ -11,75 +11,105 @@ import { skeletonStyles } from '../../styles/skeleton.js';
 
 /** @type {Scalability} */
 const SKELETON_SCALABILITY = {
-  minFlavor: { name: '??' },
-  maxFlavor: { name: '?' },
+  minFlavor: { name: '??', cpus: 0, gpus: 0, mem: 0, microservice: false, monthlyCost: 0 },
+  maxFlavor: { name: '?', cpus: 0, gpus: 0, mem: 0, microservice: false, monthlyCost: 0 },
   minInstances: 0,
   maxInstances: 0,
 };
 
 /**
+ * @typedef {import('./cc-tile-scalability.type.js').TileScalabilityState} TileScalabilityState
+ * @typedef {import('./cc-tile-scalability.type.js').TileScalabilityStateLoading} TileScalabilityStateLoading
+ * @typedef {import('./cc-tile-scalability.type.js').TileScalabilityStateLoaded} TileScalabilityStateLoaded
  * @typedef {import('../common.types.js').Scalability} Scalability
+ * @typedef {import('../common.types.js').Flavor} Flavor
  */
 
 /**
  * A "tile" component to display the current config of scalability for a given app.
  *
- * ## Details
- *
- * * When `scalability` is nullish, a skeleton screen UI pattern is displayed (loading hint).
- *
  * @cssdisplay grid
  */
 export class CcTileScalability extends LitElement {
 
   static get properties () {
     return {
-      error: { type: Boolean, reflect: true },
-      scalability: { type: Object },
+      state: { type: Object },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {boolean} Displays an error message. */
-    this.error = false;
+    /** @type {TileScalabilityState} Sets the state of the component. */
+    this.state = { type: 'loading' };
+  }
 
-    /** @type {Scalability|null} Sets the scalability config of an app with details about flavors and number of instances. */
-    this.scalability = null;
+  /**
+   * @param {string} name
+   * @returns {string}
+   * @private
+   */
+  _formatFlavorName (name) {
+    // For now, we strip the ML_ prefix from ML VMs, this may change in the future
+    return name.replace(/^ML_/, '');
   }
 
+  /**
+   * @param {Flavor} flavor
+   * @returns {string|null}
+   * @private
+   */
   _getFlavorDetails (flavor) {
     if (flavor.cpus == null) {
-      return;
+      return null;
     }
     return i18n('cc-tile-scalability.flavor-info', { ...flavor });
   }
 
-  // For now, we strip the ML_ prefix from ML VMs, this may change in the future
-  _formatFlavorName (name) {
-    return name.replace(/^ML_/, '');
+  /**
+   * @param {TileScalabilityState} state
+   * @returns {state is (TileScalabilityStateLoaded | TileScalabilityStateLoading)}
+   * @private
+   */
+  _isStateLoadedOrLoading (state) {
+    return state.type === 'loaded' || state.type === 'loading';
   }
```

Comment:
nitpick: since it's a rather simple method and to improve "reading code from just one place", I'd rather move this to a variable inside the render function.